ci: Failure in GHC 9.6 build

Looks like `diagrams` failed to provide an upper bound on `dual-tree` perhaps:
```
Build log (
Error: [Cabal-7125]
Failed to build dual-tree-0.2.3.2 (which is required by exe:grease-diagrams from grease-diagrams-0.0.0). See the build log above for details.

/home/runner/.cabal/logs/ghc-9.6.7/dual-tree-0.2.3.2-2b500df4d43171269d645918219310458572e22c0990f2f8c31b4195fff95bcd.log
):
Configuring library for dual-tree-0.2.3.2...
Preprocessing library for dual-tree-0.2.3.2...
Building library for dual-tree-0.2.3.2...
[1 of 2] Compiling Data.Tree.DUAL.Internal ( src/Data/Tree/DUAL/Internal.hs, dist/build/Data/Tree/DUAL/Internal.o, dist/build/Data/Tree/DUAL/Internal.dyn_o )

src/Data/Tree/DUAL/Internal.hs:122:49: error: [GHC-7[60](https://github.com/GaloisInc/grease/actions/runs/23744093379/job/69178901054?pr=616#step:8:61)37]
Error:     Not in scope: ‘F.unzip’
    NB: the module ‘Data.Functor’ does not export ‘unzip’.
    |
122 |   sconcat = DUALTreeU . (sconcat *** sconcat) . F.unzip . fmap unDUALTreeU
    |                                                 ^^^^^^^
Error: Process completed with exit code 1.
```
https://github.com/GaloisInc/grease/actions/runs/23744093379/job/69178901054?pr=616

--- a/cabal.project
+++ b/cabal.project
@@ -74,3 +74,6 @@ package grease
 -- See Note [Asserts] in what4
 package what4
   ghc-options: -fno-ignore-asserts
+
+-- See https://github.com/diagrams/dual-tree/issues/17
+constraints: dual-tree < 0.2.3.2

